src/Microsoft.Azure.WebJobs.Host/Tables/TableAttributeBindingProvider.cs

```diff
@@ -68,32 +62,156 @@ private TableAttributeBindingProvider(INameResolver nameResolver, IStorageAccoun
 
             return new ResolvedTableAttribute(attrResolved, client);
         }
-        
+
         public static IBindingProvider Build(INameResolver nameResolver, IConverterManager converterManager, IStorageAccountProvider accountProvider, IExtensionRegistry extensions)
         {
             var original = new TableAttributeBindingProvider(nameResolver, accountProvider, extensions);
 
             converterManager.AddConverter<JObject, ITableEntity, TableAttribute>(original.JObjectToTableEntityConverterFunc);
+            converterManager.AddConverterBuilder<object, ITableEntity, TableAttribute>(original.BuildITEConverter);
+
+            // IStorageTable --> IQueryable<ITableEntity>
+            converterManager.AddConverterBuilder<IStorageTable, TableQueryableOpenType, TableAttribute>(
+                original.QueryableConverter);
 
             var bindingFactory = new BindingFactory(nameResolver, converterManager);
-            var bindAsyncCollector = bindingFactory.BindToAsyncCollector<TableAttribute, ITableEntity>(original.BuildFromTableAttribute, null, original.CollectAttributeInfo);
 
-            var bindToJobject = bindingFactory.BindToExactAsyncType<TableAttribute, JObject>(original.BuildJObject, null, original.CollectAttributeInfo);
-            var bindToJArray = bindingFactory.BindToExactAsyncType<TableAttribute, JArray>(original.BuildJArray, null, original.CollectAttributeInfo);
+            var bindToExactCloudTable = bindingFactory.BindToGeneralAsyncType<TableAttribute, CloudTable>(
+                original.BindToCloudTable,
+                original.ToParameterDescriptorForCollector,
+                original.CollectAttributeInfo);
+
+            // Includes converter manager, which provides access to IQueryable<ITableEntity>
+            var bindToExactTestCloudTable = bindingFactory.BindToGeneralAsyncType<TableAttribute, IStorageTable>(
+                original.BindToTestCloudTable,
+                original.ToParameterDescriptorForCollector,
+                original.CollectAttributeInfo);
+
+            var bindAsyncCollector = bindingFactory.BindToAsyncCollector<TableAttribute, ITableEntity>(
+                original.BuildFromTableAttribute,
+                null,
+                original.CollectAttributeInfo);
+
+            var bindToJobject = bindingFactory.BindToExactAsyncType<TableAttribute, JObject>(
+                original.BuildJObject,
+                null,
+                original.CollectAttributeInfo);
```

Comment:
Why can't you do that in a binding rule? Is it because you don't have access to the ParameterInfo? Could that get passed into the rule as well? My concern is that we're adding a hook to our 'new' binding rules in order to maintain legacy support. And the more 'hooks' we expose, the more confused I get :-) We really need some kind of pipeline diagram to show people where they can plug in, and how.